### PR TITLE
Fix handling of coverage in sanitycheck

### PR DESCRIPTION
**Pull request by nashif**
Do no force CONFIG_CONFIG throuh CMake, use overlays instead.

### CODE REVIEW

**Review by aescolar — DISMISSED** [review]

File: scripts/sanitycheck [hunk 1]
```diff
@@ -1014,8 +1014,7 @@ class MakeGenerator:
             handler = DeviceHandler(instance)
 
         if options.enable_coverage:
-            args += ["COVERAGE=1", "EXTRA_LDFLAGS=--coverage"]
```
[the remark this answers is not in the document]
Comment (nashif) [reply scripts/sanitycheck]: fixed

**Comment by aescolar** [comment]
uhm... why is codecov still repoting the crc files not covered? (those are from unit tests)

**Comment by nashif** [comment]
> uhm... why is codecov still repoting the crc files not covered? (those are from unit tests)

when I run locally it works just fine, not sure why codecov.io is still showing those.

File: scripts/sanitycheck [hunk 2]
```diff
@@ -1041,10 +1040,13 @@ class MakeGenerator:
             by execute() will be keyed by its .name field.
         """
         args = ti.test.extra_args[:]
-        if len(ti.test.extra_configs) > 0:
+        if len(ti.test.extra_configs) > 0 or options.coverage:
             args.append("OVERLAY_CONFIG=%s" %
                         os.path.join(ti.outdir, "overlay.conf"))
 
+        if ti.test.type == "unit" and options.coverage:
+            args.append("COVERAGE=1")
```
Comment (aescolar) [inline scripts/sanitycheck]: The problem is that this path is not executed with --enable-coverage (but it is with -C)
Meaning it should be `options.enable_coverage` instead

**Review by aescolar — DISMISSED** [review]

**Review by aescolar — APPROVED** [review]
If you just did that change i pasted in the previous review, then I'm happy (I had done just that locally and it worked as expected).
Otherwise please dismiss this review and reping me

**Pull request merged**